src/interactableComponents/drawer/index.js

```diff
@@ -0,0 +1,469 @@
+  }
+  getMaxItemWidth() {
+    const {rightItems, width} = this.props;
+    return rightItems ? (width - MIN_LEFT_MARGIN) / rightItems.length : (width - MIN_LEFT_MARGIN);
+  }
+  getRightItemsTotalWidth(numberOfItems) {
+    const {rightItems} = this.props;
+    let total = 0;
+
+    if (rightItems && rightItems.length > 0) {
+      const items = rightItems.reverse();
+      const size = numberOfItems && numberOfItems >= 0 ? numberOfItems : items.length;
+      
+      for (let i = 0; i < size; i++) {
+        total += this.getItemWidth(items[i]);
+      }
+    }
+    return total;
+  }
+  getItemWidth(item) {
+    if (item && item.width) {
+      if (item.width <= this.minItemWidth) {
+        return this.minItemWidth;
+      }
+      if (item.width >= this.maxItemWidth) {
+        return this.maxItemWidth;
+      }
+      return item.width;
+    }
+    return this.minItemWidth;
+  }
+
+  getBoundaries() {
+    const {leftItem, rightItems} = this.props;
+    const rightWidth = this.getRightItemsTotalWidth();
+    const size = rightItems ? rightItems.length : 0;
+    
+    const rightBound = rightWidth > 0 ? -rightWidth : -(this.minItemWidth * size);
+    return {right: _.isEmpty(leftItem) ? 0 : this.getItemWidth(leftItem), left: _.isEmpty(rightItems) ? 0 : rightBound};
+  }
+  getSnapPoints() {
+    const {leftItem, rightItems, damping, tension} = this.props;
+    const size = rightItems ? rightItems.length : 0;
+    
+    const left = !_.isEmpty(leftItem) ? {x: this.getItemWidth(leftItem), damping: 1 - damping, tension} : {};
+    const initial = {x: 0, damping: 1 - damping, tension};
+    const last = rightItems && !_.isEmpty(rightItems[0]) ?
+      {x: -(this.getRightItemsTotalWidth()), damping: 1 - damping, tension} : {};
+
+    switch (size) {
+      case 0: 
+        return [left, initial];
+      default:
+        return [left, initial, last];
+    }
+  }
+  getAlertAreas() {
+    const {rightItems} = this.props;
+    const size = rightItems ? rightItems.length : 0;
+
+    const first = {id: 'first', influenceArea: {left: -(this.getRightItemsTotalWidth(1))}};
+    const second = {id: 'second', influenceArea: {left: -(this.getRightItemsTotalWidth(size - 1))}};
+    
+    switch (size) {
+      case 0: 
+      case 1:
+        return [];
+      case 2:
+        return [first];
+      default:
+        return [first, second];
+    }
+  }
+  getInputRanges() {
+    const {rightItems} = this.props;
+    const size = rightItems ? rightItems.length : 0;
+    const interval = 65;
+    const inputRanges = [];
+    
+    for (let i = 0; i < size; i++) {
+      const itemWidth = this.getItemWidth(rightItems[i]);
+      const end = itemWidth - (size * BLEED);
+      const min = -(itemWidth * (i + 1));
+      const max = -(end + (interval * i));
+      // const range = [-(this.minItemWidth * (i + 1)), -(end + (interval * i))];
+      inputRanges.push([min, max]);
+    }
+    return inputRanges.reverse();
+  }
+
+  renderLeftItem() {
```

Comment:
I think it will be cleaner to separate both `LeftItem` and `RightItem` into separate subcomponents.
It will make their code readable instead of having one big render method for each.